openfeature/multi/multiprovider.go

```diff
@@ -46,17 +43,25 @@ type (
 		strategyFunc       StrategyFn[FlagTypes] // used for evaluating strategies
 		logger             *slog.Logger
 		outboundEvents     chan of.Event
-		inboundEvents      chan namedEvent
 		workerGroup        sync.WaitGroup
 		shutdownFunc       context.CancelFunc
 		globalHooks        []of.Hook
 	}
 
-	// NamedProvider allows for a unique name to be assigned to a provider during a multi-provider set up.
+	// NamedProvider extends [of.FeatureProvider] by adding a unique provider name. The Name method returns
+	// the assigned provider name, while provider returns the underlying [of.FeatureProvider] instance.
+	NamedProvider interface {
+		of.FeatureProvider
+		Name() string
+		unwrap() of.FeatureProvider
+	}
```

Comment:
I suggest adding doc comments on the methods of the interface directly rather than describing the methods in the doc comment of the interface. See for example https://pkg.go.dev/log/slog#Handler